### PR TITLE
[pkg/ottl] Add `Year` converter

### PR DESCRIPTION
**Pull request by sinkingpoint**
**Description:**

This adds in a `Year` converter that takes a `time.Time` and returns the int month component. Analogous to what we already have with `Hour`

**Link to tracking Issue:** #33106

**Testing:** Added two unit tests based on the existing `Hour` ones

**Documentation:** Added the new converter to the ottlfuncs README.md

### CODE REVIEW

File: pkg/ottl/ottlfuncs/README.md [hunk 1]
```diff
@@ -1298,6 +1299,20 @@ Examples:
 
 The `UUID` function generates a v4 uuid string.
 
+### Year
+
+`Year(value)`
+
+The `Year` Converter returns the year component from the specified time. The Converter [uses the `time.Year` function](https://pkg.go.dev/time#Time.Year).
```
Comment (evan-bradley) [inline pkg/ottl/ottlfuncs/README.md:1306]:
```suggestion
The `Year` Converter returns the year component from the specified time using the Go stdlib [`time.Year` function](https://pkg.go.dev/time#Time.Year).
```

File: pkg/ottl/ottlfuncs/README.md [hunk 2]
```diff
@@ -1298,6 +1299,20 @@ Examples:
...
+
+`value` is a `time.Time`. If `value` is another type an error is returned.
```
Comment (evan-bradley) [inline pkg/ottl/ottlfuncs/README.md:1308]:
```suggestion
`value` is a `time.Time`. If `value` is another type, an error is returned.
```

**Review by evan-bradley — APPROVED** [review]

File: pkg/ottl/ottlfuncs/func_year.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	return Year(args.Time)
+}
+
+func Year[K any](duration ottl.TimeGetter[K]) (ottl.ExprFunc[K], error) {
+	return func(ctx context.Context, tCtx K) (any, error) {
+		d, err := duration.Get(ctx, tCtx)
+		if err != nil {
+			return nil, err
+		}
+		return int64(d.Year()), nil
```
Comment (TylerHelmuth) [inline pkg/ottl/ottlfuncs/func_year.go:37]: This acts differently than our `Hour` function that uses `.Hour()` on the `time.Time` struct.  Is there a reason our implementation of `Year` needs to be different?

Comment (TylerHelmuth) [reply pkg/ottl/ottlfuncs/func_year.go:37]: nvm, I got mixed up by the variable name

File: pkg/ottl/ottlfuncs/func_year.go [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+	args, ok := oArgs.(*YearArguments[K])
+
+	if !ok {
+		return nil, fmt.Errorf("YearFactory args must be of type *YearArguments[K]")
+	}
+
+	return Year(args.Time)
+}
+
+func Year[K any](duration ottl.TimeGetter[K]) (ottl.ExprFunc[K], error) {
```
Comment (TylerHelmuth) [inline pkg/ottl/ottlfuncs/func_year.go:31]:
```suggestion
func Year[K any](t ottl.TimeGetter[K]) (ottl.ExprFunc[K], error) {
```

File: pkg/ottl/ottlfuncs/func_year.go [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+	return func(ctx context.Context, tCtx K) (any, error) {
+		d, err := duration.Get(ctx, tCtx)
```
Comment (TylerHelmuth) [inline pkg/ottl/ottlfuncs/func_year.go:33]:
```suggestion
		time, err := t.Get(ctx, tCtx)
```

File: pkg/ottl/ottlfuncs/func_year.go (hunk 3 above)
Comment (TylerHelmuth) [inline pkg/ottl/ottlfuncs/func_year.go:37]:
```suggestion
		return int64(time.Year()), nil
```

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**